### PR TITLE
Fix multi scene script error and GC problem

### PR DESCRIPTION
**Pull request by GuoLei1990**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: packages/core/src/ComponentsManager.ts [hunk 1]
```diff
@@ -206,18 +200,22 @@ export class ComponentsManager {
...
+    scripts.endLoop();
   }
 
   callCameraOnEndRender(camera: Camera): void {
     const scripts = camera.entity._scripts;
+    scripts.startLoop();
     for (let i = scripts.length - 1; i >= 0; --i) {
       const script = scripts.get(i);
-      script._waitHandlingInValid || script.onEndRender(camera);
+      script?.onEndRender(camera);
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:216]:
```
scripts.get(i)?.onEndRender(camera);
```

File: packages/core/src/ComponentsManager.ts [hunk 2]
```diff
@@ -206,18 +200,22 @@ export class ComponentsManager {
 
   callCameraOnBeginRender(camera: Camera): void {
     const scripts = camera.entity._scripts;
+    scripts.startLoop();
     for (let i = scripts.length - 1; i >= 0; --i) {
       const script = scripts.get(i);
-      script._waitHandlingInValid || script.onBeginRender(camera);
+      script?.onBeginRender(camera);
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:206]: scripts.get(i)?.onBeginRender(camera);

File: packages/core/src/ComponentsManager.ts [hunk 3]
```diff
@@ -137,33 +132,42 @@ export class ComponentsManager {
...
   callScriptOnPhysicsUpdate(): void {
-    const elements = this._onPhysicsUpdateScripts._elements;
-    for (let i = this._onPhysicsUpdateScripts.length - 1; i >= 0; --i) {
+    const onPhysicsUpdateScripts = this._onPhysicsUpdateScripts;
+    onPhysicsUpdateScripts.startLoop();
+    const elements = onPhysicsUpdateScripts._elements;
+    for (let i = onPhysicsUpdateScripts.length - 1; i >= 0; --i) {
       const element = elements[i];
-      if (!element._waitHandlingInValid && element._started) {
+      if (element?._started) {
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:166]: element?._started && element.onPhysicsUpdate();

File: packages/core/src/ComponentsManager.ts [hunk 4]
```diff
@@ -137,33 +132,42 @@ export class ComponentsManager {
...
   callScriptOnLateUpdate(deltaTime: number): void {
-    const elements = this._onLateUpdateScripts._elements;
-    for (let i = this._onLateUpdateScripts.length - 1; i >= 0; --i) {
+    const onLateUpdateScripts = this._onLateUpdateScripts;
+    onLateUpdateScripts.startLoop();
+    const elements = onLateUpdateScripts._elements;
+    for (let i = onLateUpdateScripts.length - 1; i >= 0; --i) {
       const element = elements[i];
-      if (!element._waitHandlingInValid && element._started) {
+      if (element?._started) {
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:153]: element?._started && element.onLateUpdate(deltaTime);

File: packages/core/src/ComponentsManager.ts [hunk 5]
```diff
@@ -137,33 +132,42 @@ export class ComponentsManager {
...
   callScriptOnUpdate(deltaTime: number): void {
-    const elements = this._onUpdateScripts._elements;
-    for (let i = this._onUpdateScripts.length - 1; i >= 0; --i) {
+    const onUpdateScripts = this._onUpdateScripts;
+    onUpdateScripts.startLoop();
+    const elements = onUpdateScripts._elements;
+    for (let i = onUpdateScripts.length - 1; i >= 0; --i) {
       const element = elements[i];
-      if (!element._waitHandlingInValid && element._started) {
+      if (element?._started) {
```
Comment (cptbtptpbcptdtptp) [inline packages/core/src/ComponentsManager.ts:140]: element?._started && element.onUpdate(deltaTime);

File: packages/core/src/ComponentsManager.ts (hunk 3 above)
Comment (GuoLei1990) [reply packages/core/src/ComponentsManager.ts:167]: done

File: packages/core/src/ComponentsManager.ts (hunk 2 above)
Comment (GuoLei1990) [reply packages/core/src/ComponentsManager.ts]: done

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**